Library/brew.rb

```diff
@@ -4,16 +4,17 @@
 
 HOMEBREW_BREW_FILE = ENV['HOMEBREW_BREW_FILE']
 
-if ARGV == %w{--prefix}
-  puts File.dirname(File.dirname(HOMEBREW_BREW_FILE))
-  exit 0
-end
-
 require 'pathname'
 HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent.join("Homebrew")
 $:.unshift(HOMEBREW_LIBRARY_PATH.to_s)
+$:.unshift("#{HOMEBREW_LIBRARY_PATH}/test/lib") if ENV["HOMEBREW_TEST_LIB"]
 require 'global'
 
+if ARGV == %w{--prefix}
+  puts HOMEBREW_PREFIX
+  exit 0
+end
```

Comment:
I'll benchmark the performance and see how it changes. It concerns me that as-is it's using different logic to calculate the prefix which means in some weird circumstances it could produce a different result.